app/api/v1/users.rb

```diff
@@ -0,0 +1,17 @@
+class V1::Users < Grape::API
+  before do
+    authenticate!
+  end
+
+  desc 'Return all users', { tags: ['user'] }
+  params do
+    optional :created_at, type: DateTime
+  end
+  paginate
+  get '/users/' do
+    users = User.all
+    users = users.where("created_at > ?", params[:created_at]) if params[:created_at]
```

Comment:
Actually it isn't fetching until serving data. It does a relation chain, and in the end only one query will hit the database.